inc/shortcodes/class-scribd.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Shortcake_Bakery\Shortcodes;
+
+class Scribd extends Shortcode {
+
+	public static function get_shortcode_ui_args() {
+		return array(
+			'label'          => esc_html__( 'Scribd', 'shortcake-bakery' ),
+			'listItemImage'  => 'TK',
+			'attrs'          => array(
+				array(
+					'label'        => esc_html__( 'URL', 'shortcake-bakery' ),
+					'attr'         => 'url',
+					'type'         => 'text',
+					'description'  => esc_html__( 'Full URL to the Scribd document', 'shortcake-bakery' ),
+				),
+			),
+		);
+	}
+
+	public static function reversal( $content ) {
+		if ( preg_match_all( '#<iframe class="scribd_iframe_embed" src=[\'\"]([^\'\"]+)[\'\"].+data-auto-height=[\'\"][^\'\"]+[\'\"] data-aspect-ratio=[\'\"][^\'\"]+[\'\"] scrolling=[\'\"][^\'\"]+[\'\"] id=[\'\"][^\'\"]+[\'\"] width=[\'\"][^\'\"]+[\'\"] height=[\'\"][^\'\"]+[\'\"] frameborder="0"></iframe>?#', $content, $matches ) ) {
+			$replacements = array();
+			$shortcode_tag = self::get_shortcode_tag();
+			foreach ( $matches[0] as $key => $value ) {
+				$url = $matches[1][ $key ];
+				$url = explode( 'content?', $url );
+				$url = $url[0];
+				$url = str_replace( '/embeds/', '/doc/', $url );
+				$replacements[ $value ] = '[' . $shortcode_tag . ' url="' . esc_url( $url ) . '"]';
+			}
+			$content = str_replace( array_keys( $replacements ), array_values( $replacements ), $content );
+		}
+		return $content;
+	}
+
+	public static function callback( $attrs, $content = '' ) {
+
+		if ( empty( $attrs['url'] ) ) {
+			return '';
+		}
+
+		if ( ! preg_match( '#(http|https)://www\.scribd\.com/doc/?(\d)+/#', $attrs['url'], $needle ) ) {
+			if ( current_user_can( 'edit_posts' ) ) {
+				return '<div class="shortcake-bakery-error"><p>' . sprintf( esc_html__( 'Invalid Scribd URL: %s', 'shortcake-bakery' ), esc_url( $attrs['url'] ) ) . '</p></div>';
+			} else {
+				return '';
+			}
+		}
+
+		preg_match( '#\b(?:0x)?\d+\b#', $needle[0], $id );
```

Comment:
There is performance cost to using regex, so you should use non-regex alternatives if you have them.